codectrl-gui/src/components/details_view_components/code_highlighting.rs

```diff
@@ -1,94 +1,33 @@
 use egui::TextFormat;
-use std::collections::HashMap;
-use tree_sitter_highlight::{HighlightConfiguration, HighlightEvent, Highlighter};
-
-const COMMENTS: egui::Color32 = egui::Color32::from_rgb(121, 121, 121);
-const WHITE: egui::Color32 = egui::Color32::from_rgb(214, 214, 214);
-const YELLOW: egui::Color32 = egui::Color32::from_rgb(255, 216, 102);
-const GREEN: egui::Color32 = egui::Color32::from_rgb(169, 220, 118);
-const ORANGE: egui::Color32 = egui::Color32::from_rgb(255, 97, 136);
-const PURPLE: egui::Color32 = egui::Color32::from_rgb(158, 134, 200);
-const PINK: egui::Color32 = egui::Color32::from_rgb(176, 82, 121);
-const BLUE: egui::Color32 = egui::Color32::from_rgb(120, 220, 232);
-
-fn create_config(lang: &str) -> Option<HighlightConfiguration> {
-    match lang {
-        "python" => {
-            return Some(
-                HighlightConfiguration::new(
-                    tree_sitter_python::language(),
-                    tree_sitter_python::HIGHLIGHT_QUERY,
-                    "",
-                    "",
-                )
-                .unwrap(),
-            );
-        },
-
-        "rust" => {
-            return Some(
-                HighlightConfiguration::new(
-                    tree_sitter_rust::language(),
-                    tree_sitter_rust::HIGHLIGHT_QUERY,
-                    "",
-                    "",
-                )
-                .unwrap(),
-            );
-        },
-
-        _ => return None,
-    }
-}
+use syntect::{
+    easy::HighlightLines,
+    highlighting::{Style, ThemeSet},
+    parsing::SyntaxSet,
+    util::LinesWithEndings,
+};
 
 pub fn code_highlighter(code: &str, lang: &str) -> egui::text::LayoutJob {
```

Comment:
If you change `lang` to a reference of the actual `Log` itself, then you can determine the language from the file extension and/or `log.language`.